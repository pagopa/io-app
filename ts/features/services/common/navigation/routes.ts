export const SERVICES_ROUTES = {
  SERVICES_NAVIGATOR: "SERVICES_NAVIGATOR",
  SERVICES_HOME: "SERVICES_HOME",
  SERVICE_DETAIL: "SERVICE_DETAIL",
  SERVICE_WEBVIEW: "SERVICE_WEBVIEW"
} as const;
