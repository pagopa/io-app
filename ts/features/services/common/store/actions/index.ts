import { ServicesHomeActions } from "../../../home/store/actions";
import { ServiceDetailsActions } from "../../../details/store/actions/details";
import { ServicePreferenceActions } from "../../../details/store/actions/preference";

export type ServicesActions =
  | ServicesHomeActions
  | ServiceDetailsActions
  | ServicePreferenceActions;
