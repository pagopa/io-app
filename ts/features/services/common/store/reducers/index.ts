import { combineReducers } from "redux";
import homeReducer, { ServicesHomeState } from "../../../home/store/reducers";

export type ServicesState = {
  home: ServicesHomeState;
};

const servicesReducer = combineReducers({
  home: homeReducer
});

export default servicesReducer;
