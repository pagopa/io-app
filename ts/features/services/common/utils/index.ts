import { ImageURISource } from "react-native";
import { ServicePublic } from "../../../../../definitions/backend/ServicePublic";
import { contentRepoUrl } from "../../../../config";

export function getLogoForInstitution(
  institutionId: string,
  logosRepoUrl: string = `${contentRepoUrl}/logos`
) {
  return [`organizations/${institutionId.replace(/^0+/, "")}`].map(u => ({
    uri: `${logosRepoUrl}/${u}.png`
  }));
}

/**
 * Returns an array of ImageURISource pointing to possible logos for the
 * provided service.
 *
 * The returned array is suitable for being used with the MultiImage component.
 * The arrays will have first the service logo, then the organization logo.
 */
export function logosForService(
  service: ServicePublic,
  logosRepoUrl: string = `${contentRepoUrl}/logos`
): ReadonlyArray<ImageURISource> {
  return [
    `services/${service.service_id.toLowerCase()}`,
    `organizations/${service.organization_fiscal_code.replace(/^0+/, "")}`
  ].map(u => ({
    uri: `${logosRepoUrl}/${u}.png`
  }));
}
