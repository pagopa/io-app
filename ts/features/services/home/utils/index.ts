import { ImageURISource } from "react-native";
import { Institution } from "../../../../../definitions/services/Institution";
import { contentRepoUrl } from "../../../../config";

export function logoForInstitution(
  institution: Institution,
  logosRepoUrl: string = `${contentRepoUrl}/logos`
): ReadonlyArray<ImageURISource> {
  return [`organizations/${institution.fiscal_code.replace(/^0+/, "")}`].map(
    u => ({
      uri: `${logosRepoUrl}/${u}.png`
    })
  );
}
