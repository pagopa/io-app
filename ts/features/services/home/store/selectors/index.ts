import * as pot from "@pagopa/ts-commons/lib/pot";
import { pipe } from "fp-ts/lib/function";
import { createSelector } from "reselect";
import { GlobalState } from "../../../../../store/reducers/types";

export const paginatedInstitutionsPotSelector = (state: GlobalState) =>
  state.features.services.home.paginatedInstitutions;

export const paginatedInstitutionsSelector = createSelector(
  paginatedInstitutionsPotSelector,
  pot.toUndefined
);

/**
 * Returns the current page of the paginated institutions.
 *
 * | count | offset | limit | result |
 * |------:|-------:|------:|-------:|
 * | 55    | 0      | 20    | 0      |
 * | 55    | 20     | 20    | 1      |
 * | 55    | 40     | 20    | 2      |
 * | 55    | 60     | 20    | -1     |
 */
export const paginatedInstitutionsCurrentPageSelector = createSelector(
  paginatedInstitutionsPotSelector,
  paginatedInstitutions =>
    pot.getOrElse(
      pot.map(paginatedInstitutions, ({ count: total, limit, offset }) =>
        offset >= total ? -1 : offset / limit
      ),
      0
    )
);

export const paginatedInstitutionsLastPageSelector = createSelector(
  paginatedInstitutionsPotSelector,
  paginatedInstitutionsCurrentPageSelector,
  (paginatedInstitutionsPot, currentPage) =>
    pot.getOrElse(
      pot.map(
        paginatedInstitutionsPot,
        ({ count: total, limit }) => (currentPage + 1) * limit >= total
      ),
      false
    )
);

export const featuredServicesPotSelector = (state: GlobalState) =>
  state.features.services.home.featuredServices;

export const featuredServicesSelector = createSelector(
  featuredServicesPotSelector,
  featuredServicesPot =>
    pot.getOrElse(
      pot.map(
        featuredServicesPot,
        featuredServices => featuredServices.services
      ),
      []
    )
);

export const isLoadingFeaturedServicesSelector = (state: GlobalState) =>
  pipe(state, featuredServicesPotSelector, pot.isLoading);

export const isErrorFeaturedServicesSelector = (state: GlobalState) =>
  pipe(state, featuredServicesPotSelector, pot.isError);

export const featuredInstitutionsPotSelector = (state: GlobalState) =>
  state.features.services.home.featuredInstitutions;

export const featuredInstitutionsSelector = createSelector(
  featuredInstitutionsPotSelector,
  featuredInstitutionsPot =>
    pot.getOrElse(
      pot.map(
        featuredInstitutionsPot,
        featuredInstitutions => featuredInstitutions.institutions
      ),
      []
    )
);

export const isLoadingFeaturedInstitutionsSelector = (state: GlobalState) =>
  pipe(state, featuredInstitutionsPotSelector, pot.isLoading);

export const isErrorFeaturedInstitutionsSelector = (state: GlobalState) =>
  pipe(state, featuredInstitutionsPotSelector, pot.isError);
