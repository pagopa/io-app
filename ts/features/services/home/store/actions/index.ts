import { ActionType, createAsyncAction } from "typesafe-actions";
import { ScopeTypeEnum } from "../../../../../../definitions/services/ScopeType";
import { InstitutionsResource } from "../../../../../../definitions/services/InstitutionsResource";
import { NetworkError } from "../../../../../utils/errors";

export type PaginatedInstitutionsGetPayload = {
  limit: number;
  offset: number;
  search?: string;
  scope?: ScopeTypeEnum;
};

export const paginatedInstitutionsGet = createAsyncAction(
  "PAGINATED_INSTITUTIONS_GET_REQUEST",
  "PAGINATED_INSTITUTIONS_GET_SUCCESS",
  "PAGINATED_INSTITUTIONS_GET_FAILURE"
)<PaginatedInstitutionsGetPayload, InstitutionsResource, NetworkError>();

export type ServicesHomeActions = ActionType<typeof paginatedInstitutionsGet>;
