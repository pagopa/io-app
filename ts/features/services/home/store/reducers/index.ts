import * as pot from "@pagopa/ts-commons/lib/pot";
import { pipe } from "fp-ts/lib/function";
import { createSelector } from "reselect";
import { getType } from "typesafe-actions";
import { Institution } from "../../../../../../definitions/services/Institution";
import { InstitutionsResource } from "../../../../../../definitions/services/InstitutionsResource";
import { Action } from "../../../../../store/actions/types";
import { GlobalState } from "../../../../../store/reducers/types";
import { NetworkError } from "../../../../../utils/errors";
import { paginatedInstitutionsGet } from "../actions";

export type ServicesHomeState = {
  paginatedInstitutions: pot.Pot<InstitutionsResource, NetworkError>;
};

const INITIAL_STATE: ServicesHomeState = {
  paginatedInstitutions: pot.none
};

const homeReducer = (
  state: ServicesHomeState = INITIAL_STATE,
  action: Action
): ServicesHomeState => {
  switch (action.type) {
    case getType(paginatedInstitutionsGet.request):
      if (pot.isNone(state.paginatedInstitutions)) {
        return {
          ...state,
          paginatedInstitutions: pot.noneLoading
        };
      }

      return {
        ...state,
        paginatedInstitutions: pot.toUpdating(state.paginatedInstitutions, {
          count: 0,
          institutions: [],
          ...action.payload
        })
      };
    case getType(paginatedInstitutionsGet.success):
      if (action.payload.offset === 0) {
        return {
          ...state,
          paginatedInstitutions: pot.some(action.payload)
        };
      }

      const currentInstitutions = pot.getOrElse(
        pot.map(
          state.paginatedInstitutions,
          res => res.institutions as Array<Institution>
        ),
        []
      );

      return {
        ...state,
        paginatedInstitutions: pot.some({
          ...action.payload,
          institutions: [...currentInstitutions, ...action.payload.institutions]
        })
      };
    case getType(paginatedInstitutionsGet.failure):
      return {
        ...state,
        paginatedInstitutions: pot.toError(
          state.paginatedInstitutions,
          action.payload
        )
      };
  }
  return state;
};

export default homeReducer;

export const homeSelector = (state: GlobalState) =>
  state.features.services.home;

export const paginatedInstitutionsPotSelector = createSelector(
  homeSelector,
  home => home.paginatedInstitutions
);

export const paginatedInstitutionsSelector = createSelector(
  paginatedInstitutionsPotSelector,
  pot.toUndefined
);

export const isLoadingPaginatedInstitutionsSelector = (state: GlobalState) =>
  pipe(state, paginatedInstitutionsPotSelector, pot.isLoading);

export const isUpdatingPaginatedInstitutionsSelector = (state: GlobalState) =>
  pipe(state, paginatedInstitutionsPotSelector, pot.isUpdating);

export const isErrorPaginatedInstitutionsSelector = (state: GlobalState) =>
  pipe(state, paginatedInstitutionsPotSelector, pot.isError);
