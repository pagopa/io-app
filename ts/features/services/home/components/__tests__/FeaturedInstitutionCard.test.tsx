import React from "react";
import { createStore } from "redux";
import { applicationChangeState } from "../../../../../store/actions/application";
import { appReducer } from "../../../../../store/reducers";
import { GlobalState } from "../../../../../store/reducers/types";
import { renderScreenWithNavigationStoreContext } from "../../../../../utils/testWrapper";
import { SERVICES_ROUTES } from "../../../common/navigation/routes";
import {
  FeaturedInstitutionCard,
  FeaturedInstitutionCardProps
} from "../FeaturedInstitutionCard";

const testID = "FeaturedInstitutionCardTestID";

describe("FeaturedInstitutionCard", () => {
  it(`should match the snapshot`, () => {
    const component = render({
      id: "1",
      name: "### Institution ###",
      accessibilityLabel: "### Accessibility Label ###",
      onPress: () => undefined,
      testID
    });
    expect(component).toMatchSnapshot();
  });

  it(`should render card without pressable wrapper`, () => {
    const { queryByTestId } = render({
      id: "1",
      name: "### Institution ###",
      accessibilityLabel: "### Accessibility Label ###",
      testID
    });
    expect(queryByTestId(`${testID}-pressable`)).toBeNull();
  });

  it(`should match the snapshot when isNew is true`, () => {
    const component = render({
      id: "1",
      name: "### Institution ###",
      accessibilityLabel: "### Accessibility Label ###",
      onPress: () => undefined,
      isNew: true,
      testID
    });
    expect(component).toMatchSnapshot();
  });
});

function render(props: FeaturedInstitutionCardProps) {
  const globalState = appReducer(undefined, applicationChangeState("active"));
  return renderScreenWithNavigationStoreContext<GlobalState>(
    () => <FeaturedInstitutionCard {...props} />,
    SERVICES_ROUTES.SERVICES_HOME,
    {},
    createStore(appReducer, globalState as any)
  );
}
