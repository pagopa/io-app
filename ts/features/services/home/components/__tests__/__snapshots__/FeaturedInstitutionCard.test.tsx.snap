// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedInstitutionCard should match the snapshot 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "#FFFFFF",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          {
            "zIndex": 1,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          onLayout={[Function]}
          pointerEvents="box-none"
          style={null}
        >
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "#FFFFFF",
                  "borderBottomColor": "rgb(216, 216, 216)",
                  "flex": 1,
                  "shadowColor": "rgb(216, 216, 216)",
                  "shadowOffset": {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              {
                "height": 44,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                {
                  "height": 0,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "alignItems": "flex-start",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginStart": 0,
                    "opacity": 1,
                  }
                }
              />
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "justifyContent": "center",
                    "marginHorizontal": 16,
                    "maxWidth": 718,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  aria-level="1"
                  collapsable={false}
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 17,
                      "fontWeight": "600",
                    }
                  }
                >
                  SERVICES_HOME
                </Text>
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "alignItems": "flex-end",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginEnd": 0,
                    "opacity": 1,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          gestureResponseDistance={
            {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          onGestureCancel={[Function]}
          pointerEvents="box-none"
          sheetAllowedDetents="large"
          sheetCornerRadius={-1}
          sheetExpandsWhenScrolledToEdge={true}
          sheetGrabberVisible={false}
          sheetLargestUndimmedDetent="all"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              [
                {
                  "display": "flex",
                  "overflow": undefined,
                },
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              {
                "close": {
                  "animation": "spring",
                  "config": {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": {
                  "animation": "spring",
                  "config": {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                handlerTag={1}
                handlerType="PanGestureHandler"
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  {
                    "flex": 1,
                    "transform": [
                      {
                        "translateX": 0,
                      },
                      {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="box-none"
                  style={
                    [
                      {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      [
                        {
                          "backgroundColor": "#FFFFFF",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        accessibilityLabel="### Accessibility Label ###"
                        accessibilityState={
                          {
                            "busy": undefined,
                            "checked": undefined,
                            "disabled": undefined,
                            "expanded": undefined,
                            "selected": undefined,
                          }
                        }
                        accessibilityValue={
                          {
                            "max": undefined,
                            "min": undefined,
                            "now": undefined,
                            "text": undefined,
                          }
                        }
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        onTouchEnd={[Function]}
                        style={
                          {
                            "flexGrow": 1,
                          }
                        }
                        testID="FeaturedInstitutionCardTestID-pressable"
                      >
                        <View
                          style={
                            [
                              {
                                "transform": [
                                  {
                                    "scale": undefined,
                                  },
                                ],
                              },
                              {
                                "flexGrow": 1,
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              [
                                {
                                  "alignItems": "center",
                                  "backgroundColor": "#F4F5F8",
                                  "borderColor": "#E8EBF1",
                                  "borderCurve": "continuous",
                                  "borderRadius": 8,
                                  "borderWidth": 1,
                                  "flexDirection": "row",
                                  "height": 72,
                                  "paddingHorizontal": 12,
                                  "width": 702,
                                },
                                undefined,
                              ]
                            }
                            testID="FeaturedInstitutionCardTestID"
                          >
                            <View
                              style={
                                {
                                  "alignItems": "center",
                                  "flexDirection": "row",
                                }
                              }
                            >
                              <View
                                style={
                                  {
                                    "marginRight": 8,
                                  }
                                }
                              >
                                <View
                                  accessibilityIgnoresInvertColors={true}
                                  style={
                                    [
                                      {
                                        "borderColor": "rgba(14,15,19,0.1)",
                                        "borderCurve": "continuous",
                                        "borderWidth": 1,
                                        "overflow": "hidden",
                                      },
                                      {
                                        "backgroundColor": "#FFFFFF",
                                        "borderRadius": 8,
                                        "height": 44,
                                        "padding": 6,
                                        "width": 44,
                                      },
                                    ]
                                  }
                                >
                                  <View
                                    style={
                                      [
                                        {
                                          "backgroundColor": "#FFFFFF",
                                          "borderCurve": "continuous",
                                          "overflow": "hidden",
                                        },
                                        {
                                          "borderRadius": 2,
                                        },
                                      ]
                                    }
                                  >
                                    <Image
                                      accessibilityIgnoresInvertColors={true}
                                      onError={[Function]}
                                      source={
                                        {
                                          "uri": "https://assets.cdn.io.pagopa.it/logos/organizations/1.png",
                                        }
                                      }
                                      style={
                                        {
                                          "height": "100%",
                                          "resizeMode": "contain",
                                          "width": "100%",
                                        }
                                      }
                                    />
                                  </View>
                                </View>
                              </View>
                              <View
                                style={
                                  {
                                    "flexGrow": 1,
                                    "flexShrink": 1,
                                  }
                                }
                              >
                                <Text
                                  allowFontScaling={false}
                                  color="hanPurple-850"
                                  defaultColor="black"
                                  defaultWeight="Semibold"
                                  dynamicTypeRamp="headline"
                                  font="TitilliumSansPro"
                                  fontStyle={
                                    {
                                      "fontSize": 18,
                                      "lineHeight": 25,
                                    }
                                  }
                                  lineBreakMode="head"
                                  maxFontSizeMultiplier={1.25}
                                  numberOfLines={2}
                                  style={
                                    [
                                      {
                                        "fontSize": 18,
                                        "lineHeight": 25,
                                      },
                                      {
                                        "color": "#1A0744",
                                        "fontFamily": "Titillium Sans Pro",
                                        "fontStyle": "normal",
                                        "fontWeight": "600",
                                      },
                                    ]
                                  }
                                  weight="Semibold"
                                >
                                  ### Institution ###
                                </Text>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
    </View>
  </RNCSafeAreaProvider>
</View>
`;

exports[`FeaturedInstitutionCard should match the snapshot when isNew is true 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "#FFFFFF",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          {
            "zIndex": 1,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          onLayout={[Function]}
          pointerEvents="box-none"
          style={null}
        >
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "#FFFFFF",
                  "borderBottomColor": "rgb(216, 216, 216)",
                  "flex": 1,
                  "shadowColor": "rgb(216, 216, 216)",
                  "shadowOffset": {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              {
                "height": 44,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                {
                  "height": 0,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "alignItems": "flex-start",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginStart": 0,
                    "opacity": 1,
                  }
                }
              />
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "justifyContent": "center",
                    "marginHorizontal": 16,
                    "maxWidth": 718,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  aria-level="1"
                  collapsable={false}
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 17,
                      "fontWeight": "600",
                    }
                  }
                >
                  SERVICES_HOME
                </Text>
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "alignItems": "flex-end",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginEnd": 0,
                    "opacity": 1,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          gestureResponseDistance={
            {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          onGestureCancel={[Function]}
          pointerEvents="box-none"
          sheetAllowedDetents="large"
          sheetCornerRadius={-1}
          sheetExpandsWhenScrolledToEdge={true}
          sheetGrabberVisible={false}
          sheetLargestUndimmedDetent="all"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              [
                {
                  "display": "flex",
                  "overflow": undefined,
                },
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              {
                "close": {
                  "animation": "spring",
                  "config": {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": {
                  "animation": "spring",
                  "config": {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                handlerTag={3}
                handlerType="PanGestureHandler"
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  {
                    "flex": 1,
                    "transform": [
                      {
                        "translateX": 0,
                      },
                      {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="box-none"
                  style={
                    [
                      {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      [
                        {
                          "backgroundColor": "#FFFFFF",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        accessibilityLabel="### Accessibility Label ###"
                        accessibilityState={
                          {
                            "busy": undefined,
                            "checked": undefined,
                            "disabled": undefined,
                            "expanded": undefined,
                            "selected": undefined,
                          }
                        }
                        accessibilityValue={
                          {
                            "max": undefined,
                            "min": undefined,
                            "now": undefined,
                            "text": undefined,
                          }
                        }
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        onTouchEnd={[Function]}
                        style={
                          {
                            "flexGrow": 1,
                          }
                        }
                        testID="FeaturedInstitutionCardTestID-pressable"
                      >
                        <View
                          style={
                            [
                              {
                                "transform": [
                                  {
                                    "scale": undefined,
                                  },
                                ],
                              },
                              {
                                "flexGrow": 1,
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              [
                                {
                                  "alignItems": "center",
                                  "backgroundColor": "#F4F5F8",
                                  "borderColor": "#E8EBF1",
                                  "borderCurve": "continuous",
                                  "borderRadius": 8,
                                  "borderWidth": 1,
                                  "flexDirection": "row",
                                  "height": 72,
                                  "paddingHorizontal": 12,
                                  "width": 702,
                                },
                                {
                                  "backgroundColor": "#EEE8FC",
                                  "borderColor": "#DDD1F9",
                                },
                              ]
                            }
                            testID="FeaturedInstitutionCardTestID"
                          >
                            <View
                              style={
                                {
                                  "alignItems": "center",
                                  "flexDirection": "row",
                                }
                              }
                            >
                              <View
                                style={
                                  {
                                    "marginRight": 8,
                                  }
                                }
                              >
                                <View
                                  accessibilityIgnoresInvertColors={true}
                                  style={
                                    [
                                      {
                                        "borderColor": "rgba(14,15,19,0.1)",
                                        "borderCurve": "continuous",
                                        "borderWidth": 1,
                                        "overflow": "hidden",
                                      },
                                      {
                                        "backgroundColor": "#FFFFFF",
                                        "borderRadius": 8,
                                        "height": 44,
                                        "padding": 6,
                                        "width": 44,
                                      },
                                    ]
                                  }
                                >
                                  <View
                                    style={
                                      [
                                        {
                                          "backgroundColor": "#FFFFFF",
                                          "borderCurve": "continuous",
                                          "overflow": "hidden",
                                        },
                                        {
                                          "borderRadius": 2,
                                        },
                                      ]
                                    }
                                  >
                                    <Image
                                      accessibilityIgnoresInvertColors={true}
                                      onError={[Function]}
                                      source={
                                        {
                                          "uri": "https://assets.cdn.io.pagopa.it/logos/organizations/1.png",
                                        }
                                      }
                                      style={
                                        {
                                          "height": "100%",
                                          "resizeMode": "contain",
                                          "width": "100%",
                                        }
                                      }
                                    />
                                  </View>
                                </View>
                              </View>
                              <View
                                style={
                                  {
                                    "flexGrow": 1,
                                    "flexShrink": 1,
                                  }
                                }
                              >
                                <Text
                                  allowFontScaling={false}
                                  color="hanPurple-850"
                                  defaultColor="black"
                                  defaultWeight="Semibold"
                                  dynamicTypeRamp="headline"
                                  font="TitilliumSansPro"
                                  fontStyle={
                                    {
                                      "fontSize": 18,
                                      "lineHeight": 25,
                                    }
                                  }
                                  lineBreakMode="head"
                                  maxFontSizeMultiplier={1.25}
                                  numberOfLines={2}
                                  style={
                                    [
                                      {
                                        "fontSize": 18,
                                        "lineHeight": 25,
                                      },
                                      {
                                        "color": "#1A0744",
                                        "fontFamily": "Titillium Sans Pro",
                                        "fontStyle": "normal",
                                        "fontWeight": "600",
                                      },
                                    ]
                                  }
                                  weight="Semibold"
                                >
                                  ### Institution ###
                                </Text>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
